plugins/nf-azure/src/main/nextflow/cloud/azure/batch/AzBatchService.groovy

```diff
@@ -783,14 +785,37 @@ class AzBatchService implements Closeable {
         apply(() -> client.taskOperations().deleteTask(key.jobId, key.taskId))
     }
 
-    protected void cleanupJobs() {
+    protected void terminateJobs() {
+        /* 
+        We set the job to terminate when all tasks are complete rather than directly terminating, this allows Azure Batch to handle the termination for us.
+        */
+
         for( Map.Entry<TaskProcessor,String> entry : allJobIds ) {
             final proc = entry.key
             final jobId = entry.value
-            if( proc.hasErrors() ) {
-                log.debug "Preserving Azure job with error: ${jobId}"
-                continue
+
+            try {
+                log.trace "Terminating Azure job ${jobId}"
+
+                CloudJob job = apply(() -> client.jobOperations().getJob(jobId))
+                final poolInfo = job.poolInfo()
+
+                JobUpdateParameter jobParameter = new JobUpdateParameter()
+                        .withOnAllTasksComplete(OnAllTasksComplete.TERMINATE_JOB)
+                        .withPoolInfo(poolInfo)
+
+                apply(() -> client.jobOperations().updateJob(jobId, jobParameter))
```

Comment:
Instead of updating the job we should set this property when the job is created in `getOrCreateJob()`

Also, do we even need the config option? Can we just use this setting all the time?

Reply:
You would have to wait until all tasks for a job have been created, so it might be correct as is

Reply:
I think an Azure job corresponds to a Nextflow process, so you could update the job once the process is "closed" (all tasks have been created) and that would be the most eager approach. But might be overkill. Let's see what Paolo thinks.

Reply:
It doesn't look like jobs are re-used across runs.

We just need an "onProcessClose" event to notify when all tasks are created. I've already added it in a number of PRs, like this one: #3818

So if Paolo likes the idea then we can just rip the code from that PR.

Reply:
https://github.com/nextflow-io/nextflow/pull/3818/files#diff-3250bcd11fe7bb7aeeaf9839c44ac5d737dc27ddc9a734954789d09eb9dab399

Reply:
Make sense. It can be improved in a following iteration